src/modules/adaptors/Morpho/MorphoAaveV2ATokenAdaptor.sol

```diff
@@ -0,0 +1,171 @@
+    // and allow user withdraws from aToken positions that are backing loans.
+    // This will be mitigated by only allowing trusted strategists to use this adaptor,
+    // and educating them the dangers of misconfiguring their position.
+    //====================================================================
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Aave V2 aToken Adaptor V 1.1"));
+    }
+
+    /**
+     * @notice The Morpho Aave V2 contract on Ethereum Mainnet.
+     */
+    function morpho() internal pure returns (IMorpho) {
+        return IMorpho(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0);
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice Cellar must approve Morpho to spend its assets, then call supply to lend its assets.
+     * @param assets the amount of assets to lend on Morpho
+     * @param adaptorData adaptor data containing the abi encoded aToken
+     * @dev configurationData is not used for deposits bc it only influences user withdraws
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        // Deposit assets to Morpho.
+        IAaveToken aToken = abi.decode(adaptorData, (IAaveToken));
+        ERC20 underlying = ERC20(aToken.UNDERLYING_ASSET_ADDRESS());
+        underlying.safeApprove(address(morpho()), assets);
+
+        morpho().supply(address(aToken), assets);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(underlying, address(morpho()));
+    }
+
+    /**
+     * @notice Cellars must withdraw from Morpho
+     * @dev Important to verify that external receivers are allowed if receiver is not Cellar address.
+     * @param assets the amount of assets to withdraw from Morpho
+     * @param receiver the address to send withdrawn assets to
+     * @param adaptorData adaptor data containing the abi encoded aToken
+     * @param configData abi encoded bool indicating whether the position is liquid or not
+     */
+    function withdraw(
+        uint256 assets,
+        address receiver,
+        bytes memory adaptorData,
+        bytes memory configData
+    ) public override {
+        // Run external receiver check.
+        _externalReceiverCheck(receiver);
+
+        // Make sure position is setup to be liquid.
+        bool isLiquid = abi.decode(configData, (bool));
+        if (!isLiquid) revert BaseAdaptor__UserWithdrawsNotAllowed();
```

Comment:
Doesn't Morpho have the `fallback` setup so it switches out P2P arrangements when one party exits while capital is still deployed? Essentially where another participant is matched, or worst case the underlying pool fills in when a party leaves?

Reply:
Or is this a design intent for the Adaptors here?

Reply:
This comment can be disregarded after discussing things further. I believe you will be adding in extra comments to help clarify what you mean by the misconfigured config data.